iOS: no matching function for call to 'fieldForOffset' (builtin.cpp)

I get the following error when trying to compile the latest hello-iOS example using both avian and hello-ios from current master branch:

compiling build/ios-arm-bootimage/builtin.o
src/builtin.cpp:911:30: error: no matching function for call to 'fieldForOffset'
  ACQUIRE_FIELD_FOR_WRITE(t, fieldForOffset(t, target, offset));
                             ^~~~~~~~~~~~~~
src/avian/machine.h:45:53: note: expanded from macro 'ACQUIRE_FIELD_FOR_WRITE'
  FieldWriteResource MAKE_NAME(monitorResource_)(t, field)
                                                    ^
src/builtin.cpp:77:10: note: candidate function not viable: cannot convert from
      base class pointer 'object' (aka 'vm::GcObject _') to derived class
      pointer 'vm::GcSingleton *' for 2nd argument
GcField_ fieldForOffset(Thread\* t, GcSingleton\* o, unsigned offset)
         ^
1 error generated.
make[1]: **\* [build/ios-arm-bootimage/builtin.o] Error 1
make: **\* [make-vm] Error 2

Cheers
Ben

--- a/src/builtin.cpp
+++ b/src/builtin.cpp
@@ -74,11 +74,16 @@ GcField* fieldForOffsetInClass(Thread* t, GcClass* c, unsigned offset)
   return 0;
 }
 
-GcField* fieldForOffset(Thread* t, GcSingleton* o, unsigned offset)
+GcField* fieldForOffset(Thread* t, object o, unsigned offset)
 {
   GcClass* c = objectClass(t, o);
   if (c->vmFlags() & SingletonFlag) {
-    c = cast<GcClass>(t, singletonObject(t, o, 0));
+    GcSingleton* s = cast<GcSingleton>(t, o);
+
+    // If the object is a Singleton, we assume it's the static table of a class -
+    // which will always have the parent class as the first (0th) element.
+    c = cast<GcClass>(t, singletonObject(t, s, 0));
+
     object table = c->fieldTable();
     if (table) {
       for (unsigned i = 0; i < objectArrayLength(t, table); ++i) {

